Add support for `<iframe>` in Layout 2020

mrobinson opened the pull request:
<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mrobinson commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit e285253a4106b5839af877dc78faf71abb97abb2 with merge 9e99795ce04bcd1ad27169e83611d3db50db4b3e...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/actions/runs/4425799323/jobs/7763179656)

mrobinson committed:
Add support for <iframe> elements for Layout 2020

This change adds support for the <iframe> element to Layout 2020. In
addition, certain aspects of the implementation are made the same
between both layout systems.

mrobinson commented:
@bors-servo try

delan reviewed (approved):
Great work! The impl looks good, though it might be worth investigating why some of the siblings of the newly passing tests that also involve iframes are still failing:

* /css/CSS2/floats-clear/float-replaced-height-005.xht
* /css/CSS2/normal-flow/block-replaced-height-005.xht
* /css/CSS2/normal-flow/inline-block-replaced-height-005.xht
* /css/CSS2/normal-flow/inline-replaced-height-005.xht
* /css/CSS2/positioning/absolute-replaced-height-{005,007,012,014,019,021,026,028,033,035}.xht

delan commented:
Nice, thanks for looking into that!

@bors-servo r+

bors-servo commented:
:pushpin: Commit 9e0b41e has been approved by `delan`

<!-- @bors-servo r=delan 9e0b41ebc425701e44799b9aaff1c41a65b6b55a -->

bors-servo commented:
:hourglass: Testing commit 9e0b41ebc425701e44799b9aaff1c41a65b6b55a with merge a64a15ba5303aa52b21b6079ea8de430095e7caa...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/4614679976/jobs/8159220046)
Approved by: delan
Pushing a64a15ba5303aa52b21b6079ea8de430095e7caa to master...
